plugins/epochstorage/plugin.go

```diff
@@ -0,0 +1,474 @@
+}
+
+func removeMessageFromEpoch(ei epoch.Index, msgID tangle.MessageID) error {
+	epochContentStorage, err := getEpochContentStorage(ei)
+	if err != nil {
+		return err
+	}
+
+	if err := epochContentStorage.messageIDs.Delete(msgID.Bytes()); err != nil {
+		return errors.New("Fail to remove Message from epoch store")
+	}
+	return nil
+}
+
+func insertTransactionToEpoch(ei epoch.Index, txID utxo.TransactionID) error {
+	epochContentStorage, err := getEpochContentStorage(ei)
+	if err != nil {
+		return err
+	}
+
+	if err := epochContentStorage.transactionIDs.Set(txID.Bytes(), txID.Bytes()); err != nil {
+		return errors.New("Fail to insert Transaction to epoch store")
+	}
+	return nil
+}
+
+func removeTransactionFromEpoch(ei epoch.Index, txID utxo.TransactionID) error {
+	epochContentStorage, err := getEpochContentStorage(ei)
+	if err != nil {
+		return err
+	}
+
+	if err := epochContentStorage.transactionIDs.Delete(txID.Bytes()); err != nil {
+		return errors.New("Fail to remove Transaction from epoch store")
+	}
+	return nil
+}
+
+func insertOutputsToEpoch(ei epoch.Index, spent, created []*ledger.OutputWithMetadata) error {
+	epochContentStorage, err := getEpochContentStorage(ei)
+	if err != nil {
+		return err
+	}
+
+	for _, s := range spent {
+		if err := epochContentStorage.spentOutputs.Set(s.ID().Bytes(), s.ID().Bytes()); err != nil {
+			return errors.New("Fail to insert spent output to epoch store")
+		}
+	}
+
+	for _, c := range created {
+		if err := epochContentStorage.createdOutputs.Set(c.ID().Bytes(), c.ID().Bytes()); err != nil {
+			return errors.New("Fail to insert created output to epoch store")
+		}
+	}
+
+	return nil
+}
+
+func removeOutputsFromEpoch(ei epoch.Index, spent, created []*ledger.OutputWithMetadata) error {
+	epochContentStorage, err := getEpochContentStorage(ei)
+	if err != nil {
+		return err
+	}
+
+	for _, s := range spent {
+		if err := epochContentStorage.spentOutputs.Delete(s.ID().Bytes()); err != nil {
+			return errors.New("Fail to remove spent output from epoch store")
+		}
+	}
+
+	for _, c := range created {
+		if err := epochContentStorage.createdOutputs.Delete(c.ID().Bytes()); err != nil {
+			return errors.New("Fail to remove created output from epoch store")
+		}
+	}
+
+	return nil
+}
+
+func saveEpochVotersWeight(message *tangle.Message) {
+	voter := identity.NewID(message.IssuerPublicKey())
+	activeWeights, _ := deps.Tangle.WeightProvider.WeightsOfRelevantVoters()
+
+	epochVotersWeightMutex.Lock()
+	defer epochVotersWeightMutex.Unlock()
+	epochIndex := message.M.EI
+	if _, ok := epochVotersWeight[epochIndex]; !ok {
+		epochVotersWeight[epochIndex] = make(map[epoch.ECR]map[identity.ID]float64)
+	}
+	if _, ok := epochVotersWeight[epochIndex][message.M.ECR]; !ok {
+		epochVotersWeight[epochIndex][message.M.ECR] = make(map[identity.ID]float64)
+	}
+	epochVotersWeight[epochIndex][message.M.ECR][voter] = activeWeights[voter]
+}
+
+func outputIDFromBytes(outputBytes []byte) (utxo.OutputID, error) {
```

Comment:
This can be useful in general. Let's put it in utxo/types.go next to the other OutputID From functions